homework-functions/index.js

```diff
@@ -0,0 +1,42 @@
+module.exports = createAutoComplete;
+
+// First implementation of autocomplete
+
+function createAutoComplete(array) {
```

Comment:
So technically yes, it will work :) But it's really not the best solution. So the complexity is about O(N).
I suggest you create something like [this](https://en.wikipedia.org/wiki/Trie).